parachain/pallets/teebag/src/lib.rs

```diff
@@ -700,21 +713,41 @@ impl<T: Config> Pallet<T> {
 		Ok(().into())
 	}
 
-	fn add_enclave_identifier(worker_type: WorkerType, who: &T::AccountId) -> Result<(), Error<T>> {
+	pub fn add_enclave_identifier_internal(
+		worker_type: WorkerType,
+		who: &T::AccountId,
+	) -> Result<(), Error<T>> {
 		EnclaveIdentifier::<T>::try_mutate(worker_type, |v| {
 			ensure!(!v.contains(who), Error::<T>::EnclaveIdentifierAlreadyExist);
 			v.try_push(who.clone()).map_err(|_| Error::<T>::MaxEnclaveIdentifierOverflow)
 		})
 	}
 
 	pub fn add_enclave(sender: &T::AccountId, enclave: &Enclave) -> Result<(), Error<T>> {
-		match EnclaveRegistry::<T>::get(sender) {
-			Some(old_enclave) => ensure!(
-				old_enclave.worker_type == enclave.worker_type,
-				Error::<T>::UnexpectedWorkerType
-			),
-			None => Self::add_enclave_identifier(enclave.worker_type, sender)?,
+		match Self::mode() {
```

Comment:
How about removing the duplicated check of the worker_type? something like:

```rust
if let Some(old_enclave) = EnclaveRegistry::<T>::get(sender) {
    ensure!(old_enclave.worker_type == enclave.worker_type, Error::<T>::UnexpectedWorkerType);
}

match Self::mode() {
    OperationalMode::Production | OperationalMode::Maintenance => {
        ensure!(
            EnclaveIdentifier::<T>::get(enclave.worker_type).contains(sender),
	    Error::<T>::EnclaveIdentifierNotExist
         );
    },
    OperationalMode::Development => {
        if EnclaveRegistry::<T>::get(sender).is_none() {
	    Self::add_enclave_identifier_internal(enclave.worker_type, sender)?;
	}
    },
};
```
it would also make the logic easier to read